conbench/entities/commit.py

```diff
@@ -172,6 +180,62 @@ def get_github_commit(repository: str, pr_number: str, branch: str, sha: str) ->
     return commit
 
 
+def backfill_default_branch_commits(
+    repository: str, new_commit: Commit
+) -> List[Commit]:
+    """Catches up the default-branch commits in the database.
+
+    Will search GitHub for any untracked commits, between the given new_commit back in
+    time to the last tracked commit, and backfill all of them.
+
+    Won't backfill any commits before the last tracked commit. But if there are no
+    commits in the database, will backfill them all.
+    """
+    github = GitHub()
+    name = repository_to_name(repository)
+    default_branch = github.get_default_branch(name)
+
+    last_tracked_commit = Commit.all(
+        filter_args=[Commit.sha != new_commit.sha, Commit.timestamp.isnot(None)],
+        branch=default_branch,
+        repository=repository,
+        order_by=Commit.timestamp.desc(),
+        limit=1,
+    )
+
+    if last_tracked_commit:
+        since = last_tracked_commit[0].timestamp
+    elif os.getenv("DB_HOST") == "localhost":
+        # only for conbench/tests/populate_local_conbench.py
+        since = datetime(2022, 1, 1)
```

Comment:
There are other cases where the DB will be on localhost, though the majority will still be testing of other sorts. Most things will get caught by the `if` clause here anyway so I'm not that worried, but maybe a log/warning in these other clauses so somebody doesn't end up in a debugging rabbit hole in the future?

Reply:
Agreed, a warning or other kind of messaging would be helpful here

Reply:
Great idea

Reply:
@ElenaHenderson if `os.getenv("DB_HOST") == "localhost"`, is it a safe assumption that someone is testing on a local computer? If not, is there a better way to check that?

Reply:
Tests and the app that are run using docker-compose will use DB_HOST = "db".
Anything in production will run on k8s pod and this is how you can check if a process is running in the pod:
`os.getenv("KUBERNETES_SERVICE_HOST") is not None`

People can set DB_HOST to `127.0.0.1` when they run postgres locally.